libs/core/src/schemas/events.schemas.ts

```diff
@@ -46,3 +46,52 @@ export const DiscordMessageCreated = z.object({
     z.literal('comment-create'),
   ]),
 });
+
+// All events should carry this common metadata
+export const EventMetadata = z.object({
+  createdAt: z.date().describe('When the event was emitted'),
+  // TODO: TBD
+  // aggregateType: z.enum(Aggregates).describe("Event emitter aggregate type")
+  // aggregateId: z.string().describe("Event emitter aggregate id")
+  // correlation: z.string().describe("Event correlation key")
+  // causation: z.object({}).describe("Event causation")
+});
+
+// on-chain contest manager events
+export const ContestManagerDeployed = EventMetadata.extend({
+  namespace: z.string().describe('Community namespace'),
+  contest: z.string().describe('Contest manager address'),
+  interval: z.number().int().positive().describe('Recurring constest interval'),
+  oneOff: z.boolean().describe('Flags one-off contests'),
+}).describe('When a contest contract manager gets deployed');
+
+export const ContestStarted = EventMetadata.extend({
```

Comment:
For the recurring contest event NewContestStarted return the following values: uint256 indexed contestId, uint256 startTime, uint256 endTime

And for the single contest event NewContestStarted return the following values: uint256 startTime, uint256 endTime.

So we will need to give a default value for the single contest or make the contest id not required.